pages/admin/members/import.tsx

```diff
@@ -0,0 +1,236 @@
+/* eslint-disable react/jsx-props-no-spreading */
+
+import React, { useMemo, useState } from "react";
+import Papa from "papaparse";
+import _ from "lodash";
+import { useTable } from "react-table";
+import { useMutation } from "@apollo/react-hooks";
+import Layout from "layouts/admin";
+import { ServerSideProps } from "utils/getServerSideProps";
+import {
+  Level,
+  Table,
+  Button,
+  Section,
+  Container,
+  Form,
+} from "react-bulma-components";
+import importPeopleMutation from "apollo/queries/person/importPeople.gql";
+
+export { getServerSideProps } from "utils/getServerSideProps";
+
+const { InputFile } = Form;
+
+const Members = ({ user }: ServerSideProps): React.ReactElement => {
+  const [
+    importPeople,
+    { loading: importPeopleMutationLoading, error: importPeopleMutationError },
+  ] = useMutation(importPeopleMutation);
+
+  const [membersData, setMembersData] = useState<
+    { members: Record<string, unknown>[] } | undefined
+  >(undefined);
+
+  const tableColumns = useMemo(
+    () => [
+      {
+        Header: "ID",
+        accessor: "id",
+      },
+      {
+        Header: "SID",
+        accessor: "sid",
+      },
+      {
+        Header: "Chinese Name",
+        accessor: "chineseName",
+      },
+      {
+        Header: "English Name",
+        accessor: "englishName",
+      },
+      {
+        Header: "Gender",
+        accessor: "gender",
+      },
+      {
+        Header: "Date of Birth",
+        accessor: "dateOfBirth",
+      },
+      {
+        Header: "Email",
+        accessor: "email",
+      },
+      {
+        Header: "Phone",
+        accessor: "phone",
+      },
+      {
+        Header: "College",
+        accessor: "college",
+        /* Cell: ({ value }: CellProps<Record<string, unknown>, College>) => {
+          return <div>{`${value.code}`}</div>;
+        }, */
+      },
+      {
+        Header: "Major",
+        accessor: "major",
+        /* Cell: ({ value }: CellProps<Record<string, unknown>, Major>) => {
+          return <div>{`${value.code}`}</div>;
+        }, */
+      },
+      {
+        Header: "Date of Entry",
+        accessor: "dateOfEntry",
+      },
+      {
+        Header: "Expected Graduation Date",
+        accessor: "expectedGraduationDate",
+      },
+      {
+        Header: "Member Since",
+        accessor: "memberSince",
+      },
+    ],
+    []
+  );
+
+  const tableData = useMemo(() => {
+    return membersData?.members ?? [];
+  }, [membersData]);
+
+  const tableGetRowId = useMemo(() => {
+    return (row: Record<string, unknown>) => (row.id as number).toString();
+  }, []);
+
+  const tableInstance = useTable({
+    columns: tableColumns,
+    data: tableData,
+    getRowId: tableGetRowId,
+  });
+
+  const upload = () => {
+    // remove member id to prevent data collision
+    const members = membersData?.members.map((member) => _.omit(member, "id"));
+    _.chunk(members, 10).forEach((people) =>
```

Comment:
why need chunk?

Reply:
1 member 1 request, why not

Reply:
_.chunk(members, members * 10).forEach((people) =>

Reply:
you on9?